src/com/geekhub/hw2/vehicle/SolarCar.java

```diff
@@ -0,0 +1,28 @@
+package com.geekhub.hw2.vehicle;
+
+import com.geekhub.hw2.parts.EnergySource;
+import com.geekhub.hw2.parts.Engine;
+import com.geekhub.hw2.parts.Gear;
+
+public class SolarCar extends Car {
+    public SolarCar(EnergySource energySource, Engine engine, Gear gear) {
```

Comment:
In this case better to pass concrete implementation as pararameters. 
E.g. Now I can do something like that: "new SolarCar(new GasTank(), new GasolineEngine(), new BoatGear());"